libsolidity/ast/Types.cpp

```diff
@@ -44,6 +44,71 @@ using namespace std;
 using namespace dev;
 using namespace dev::solidity;
 
+namespace
+{
+
+// checks whether (_base ^ _exp) fits into 4096 bits
```

Comment:
I think `**` would be better here. Also we generally use `///` ("javadoc") to document functions.